Remove restricted sessions

jakule opened the pull request:
Closes https://github.com/gravitational/teleport/issues/36099

changelog: Remove restricted sessions for SSH

jakule committed:
Remove restricted sessions

jakule committed:
Remove missing reference to restricted session

zmb3 commented:
We should forego the short changelog entry and instead manually update CHANGELOG.md with a longer description.

jakule committed:
Update CHANGELOG.md to remove outdated SSH feature

The commit removes the reference to the deprecated and removed SSH restricted sessions feature from the CHANGELOG.md file. The restricted session feature had been phased out since Teleport 14, and this update reflects its removal in Teleport 15.

zmb3 commented on CHANGELOG.md:
```diff
@@ -72,6 +72,11 @@ by `insecure-drop`, which still creates temporary users but does not create a
+#### Remove restricted sessions for SSH
+
+The restricted session feature for SSH has been deprecated since Teleport 14 and 
+has been removed in Teleport 15. There is no replacement for this feature.
```
Instead of saying "there is no replacement" I think we should say something like:

"We recommend implementing network restrictions outside of Teleport (iptables, security groups, etc)"

zmb3 commented:
/excludeflake *

jakule committed:
Rephrase the changelog note that it's recommends the implementation of network restrictions outside of Teleport like iptables and security groups.
